stl/inc/random

```diff
@@ -263,7 +288,38 @@ _NODISCARD _Real generate_canonical(_Gen& _Gx) { // build a floating-point value
     return _Ans / _Factor;
 }
 
-#define _NRAND(eng, resty) (_STD generate_canonical<resty, static_cast<size_t>(-1)>(eng))
+template <class _Real, class _Gen>
+_NODISCARD _Real _Nrand_impl(_Gen& _Gx) { // build a floating-point value from random sequence
+    _RNG_REQUIRE_REALTYPE(_Nrand_impl, _Real);

```

Comment:
This function should be marked as `TRANSITION, for tr1 machinery only` or something like that, as it's outside the deprecated `tr1` namespace. The comment should indicate that the Standard approach would be to call `generate_canonical` directly.